src/OrleansRuntime/Catalog/ActivationData.cs

```diff
@@ -483,26 +483,38 @@ public int WaitingCount
             }
         }
 
+        public enum EnqueueMessageResult
+        {
+            Success,
+            ErrorInvalidActivation,
+            ErrorStuckActivation,
+        }
+
         /// <summary>
         /// Insert in a FIFO order
         /// </summary>
         /// <param name="message"></param>
-        public bool EnqueueMessage(Message message)
+        public EnqueueMessageResult EnqueueMessage(Message message)
         {
             lock (this)
             {
                 if (State == ActivationState.Invalid)
                 {
                     logger.Warn(ErrorCode.Dispatcher_InvalidActivation,
                         "Cannot enqueue message to invalid activation {0} : {1}", this.ToDetailedString(), message);
-                    return false;
+                    return EnqueueMessageResult.ErrorInvalidActivation;
                 }
-                // If maxRequestProcessingTime is never set, then we will skip this check
-                if (maxRequestProcessingTime.TotalMilliseconds > 0 && Running != null)
+                if (Running != null)
                 {
-                    // Consider: Handle long request detection for reentrant activations -- this logic only works for non-reentrant activations
                     var currentRequestActiveTime = DateTime.UtcNow - currentRequestStartTime;
-                    if (currentRequestActiveTime > maxRequestProcessingTime)
+                    if (currentRequestActiveTime > CollectionAgeLimit)
```

Comment:
I concur - better define a separate config parameter and default it to CollectionAgeLimit.